packages/driver/src/cypress/cy.ts

```diff
@@ -504,20 +504,38 @@ export class $Cy implements ITimeouts, IStability, IAssertions, IRetries, IJQuer
         // about:blank in a visit, we do need these
         this.contentWindowListeners(autWindow)
 
-        this.Cypress.action('app:window:load', this.state('window'))
-
-        // we are now stable again which is purposefully
-        // the last event we call here, to give our event
-        // listeners time to be invoked prior to moving on
-        this.isStable(true, 'load')
+        try {
+          this.Cypress.action('app:window:load', this.state('window'))
+        } catch (err) {
+          err.isFromWindowLoadEvent = true
+
+          throw err
+        } finally {
+          // we are now stable again which is purposefully
+          // the last event we call here, to give our event
+          // listeners time to be invoked prior to moving on
+          this.isStable(true, 'load')
+        }
       } catch (err) {
+        if (err.isFromWindowLoadEvent) {
+          delete err.isFromWindowLoadEvent
+
+          // the user's window:load handler threw an error, so propagate that
+          // and fail the test
+          const r = this.state('reject')
+
+          if (r) {
+            return r(err)
+          }
+        }
+
         // we failed setting the remote window props which
         // means the page navigated to a different domain
 
         // we expect a cross-origin error and are setting things up
         // elsewhere to handle running cross-domain, so don't fail
         // because of it
-        if (this.state('anticipatingMultidomain')) {
+        if (this.state('readyForMultidomain')) {
```

Comment:
I think this was handled in the finally block above

Reply:
I think this will end up still being necessary when I refactor to the above to not move to stable when it's a cross-origin error.